src/journal-remote/journal-remote-write.c

```diff
@@ -32,13 +32,13 @@ int writer_new(RemoteServer *server, Writer **ret) {
         if (!w)
                 return -ENOMEM;
 
+        w->n_ref = 1;
         w->metrics = server->metrics;
```

Comment:
Since you're touching this code, could you also rewrite it to use structured initialization for `w`? I.e., change `new0` to `new` and then do
```c
*w = (Writer) {
        .metrics = server->metrics,
        ...
};
```